File: backend/internal/models/rizz.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

// RizzResponse represents a generated AI response
type RizzResponse struct {
	ID          uuid.UUID      `gorm:"type:uuid;default:gen_random_uuid();primaryKey" json:"id"`
	UserID      uuid.UUID      `gorm:"type:uuid;index" json:"user_id"`
	InputText   string         `gorm:"type:text;not null" json:"input_text"`
	Tone        string         `gorm:"type:varchar(30);not null" json:"tone"`
	Response1   string         `gorm:"type:text" json:"response_1"`
	Response2   string         `gorm:"type:text" json:"response_2"`
	Response3   string         `gorm:"type:text" json:"response_3"`
	SelectedIdx int            `gorm:"default:-1" json:"selected_idx"`
	Category    string         `gorm:"type:varchar(30)" json:"category"` // dating, work, casual, family
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}

// RizzStreak tracks daily usage
type RizzStreak struct {
	ID            uuid.UUID `gorm:"type:uuid;default:gen_random_uuid();primaryKey" json:"id"`
	UserID        uuid.UUID `gorm:"type:uuid;uniqueIndex" json:"user_id"`
	CurrentStreak int       `gorm:"default:0" json:"current_streak"`
	LongestStreak int       `gorm:"default:0" json:"longest_streak"`
	TotalRizzes   int       `gorm:"default:0" json:"total_rizzes"`
	FreeUsesToday int       `gorm:"default:0" json:"free_uses_today"`
	LastUseDate   time.Time `json:"last_use_date"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

// Tone options
var RizzTones = []string{
	"flirty",
	"professional",
	"funny",
	"chill",
	"savage",
	"romantic",
	"confident",
	"mysterious",
}

// Category options
var RizzCategories = []string{
	"dating",
	"work",
	"casual",
	"family",
	"friends",
}

File: types/rizz.ts
export interface RizzResponse {
  id: string;
  input_text: string;
  tone: string;
  category: string;
  response_1: string;
  response_2: string;
  response_3: string;
  selected_idx: number;
  created_at: string;
}

export interface RizzStats {
  current_streak: number;
  longest_streak: number;
  total_rizzes: number;
  free_uses_today: number;
}

export const TONES = [
  { id: 'flirty', label: 'Flirty', emoji: '😏', color: '#ec4899' },
  { id: 'professional', label: 'Professional', emoji: '💼', color: '#3b82f6' },
  { id: 'funny', label: 'Funny', emoji: '😂', color: '#fbbf24' },
  { id: 'chill', label: 'Chill', emoji: '😎', color: '#22c55e' },
  { id: 'savage', label: 'Savage', emoji: '💅', color: '#ef4444' },
  { id: 'romantic', label: 'Romantic', emoji: '❤️', color: '#f43f5e' },
  { id: 'confident', label: 'Confident', emoji: '💪', color: '#8b5cf6' },
  { id: 'mysterious', label: 'Mysterious', emoji: '🌙', color: '#1f2937' },
];

export const CATEGORIES = [
  { id: 'dating', label: 'Dating', emoji: '💕' },
  { id: 'work', label: 'Work', emoji: '💼' },
  { id: 'casual', label: 'Casual', emoji: '💬' },
  { id: 'family', label: 'Family', emoji: '👨‍👩‍👧' },
  { id: 'friends', label: 'Friends', emoji: '👯' },
];
